packages/extism-tools/src/ExtismToolFactory.ts

```diff
@@ -0,0 +1,85 @@
+import createPlugin, { type Plugin } from "@extism/extism";
+import type { JSONSchemaType } from "ajv";
+import type { BaseTool, ToolMetadata } from "llamaindex";
+
+export const WASM_DIRECTORY = "./dist/wasm";
+export const DEFAULT_MAX_HTTP_RESPONSE_BYTES = 100 * 1024 * 1024; // 100 MB
+
+export type ToolParams = Record<string, any>;
+
+export type ToolClassParams = {
+  metadata?: ToolMetadata<JSONSchemaType<ToolParams>>;
+};
+
+export type CreateToolClassParams = {
+  wasmFilename: string;
+  allowedHosts: string[];
+  maxHttpResponseBytes: number;
+  transformResponse: (response: any) => any;
+};
+
+export enum ExtismTool {
+  WIKI = "wiki",
+}
+
+export const ToolMap: Record<`${ExtismTool}`, CreateToolClassParams> = {
+  [ExtismTool.WIKI]: {
+    wasmFilename: "wiki.wasm",
+    allowedHosts: ["*.wikipedia.org"],
+    maxHttpResponseBytes: DEFAULT_MAX_HTTP_RESPONSE_BYTES,
+    transformResponse: (response: any) => response.extract,
+  },
```

Comment:
1. can we move these default paramters to the WASM? (Similar to `getMetadata`)
2. The user can overwrite these parameters - should not be part of this factory. The factory should be tool code agnostic

Reply:
then how about we use per default:
1. we allow all hosts 
2. we use xMB memory (do we need 100MB?)
3. we use `response.extract` 
all this can be changed in `createToolClass` call if needed

Reply:
or better, we have a `${toolName}.config` file that contains these parameters and it is loaded with `createToolClass`